azurelinuxagent/common/utils/shellutil.py

```diff
@@ -117,30 +117,51 @@ def _encode_command_output(output):
     return ustr(output, encoding='utf-8', errors="backslashreplace")
 
 
-def run_command(command):
+class CommandError(Exception):
     """
-    Wrapper for subprocess.Popen. Executes the given command and returns its stdout.
-    Logs any errors executing the command and raises an exception.
+    Exception raised by run_command when the command returns an error
     """
-    retcode = 0
+    @staticmethod
+    def _get_message(command, returncode):
+        command_name = command[0] if isinstance(command, list) and len(command) > 0 else command
```

Comment:
No, the parameters may contain sensitive data that we do not want in the exception. The caller knows the command and whether it is appropriate to include the parameters or not; it can always extend the exception with the parameters if needed.